scrapy/core/scraper.py

```diff
@@ -146,7 +146,8 @@ def _scrape2(self, request_result, request, spider):
                 self._log_download_errors, request_result, request, spider)
 
     def call_spider(self, result, request, spider):
-        result.request = request
+        if getattr(result, "request", None) is None:
+            result.request = request
```

Comment:
I reviewed the surrounding code, and it looks like we can assume that `result` here is a `Response` object. So maybe a simpler change, like the one in `engine.py`, would work the same.

I would also suggest renaming `result` to `response` if it were not a backward-incompatible change :expressionless:

Reply:
Actually, `result` can be a `Failure` object, that's what the attribute check is needed: https://github.com/scrapy/scrapy/pull/4618/commits/3992338573ba64f8c7ebaf5cee919068fcdd2ea9#diff-d12d4e527e55eef105fcf232ef856619R80-R83